compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -881,6 +873,74 @@ public void visit(BLangSimpleVariable varNode) {
+        List<String> errors = new ArrayList<>();
+        if (!isSupportedConfigType(type, errors, varName) || !errors.isEmpty()) {
+            StringBuilder errorMsg = new StringBuilder();
+            for (String error : errors) {
+                errorMsg.append("\n\t").append(error);
+            }
+            dlog.error(location, DiagnosticErrorCode.CONFIGURABLE_VARIABLE_CURRENTLY_NOT_SUPPORTED, type, errorMsg);
+        }
+    }
+
+    private boolean isSupportedConfigType(BType type, List<String> errors, String varName) {
+        switch (type.getKind()) {
+            case FINITE:
+                return ((BFiniteType) type).isAnyData;
+            case ARRAY:
+                BType elementType = ((BArrayType) type).eType;
+                if (elementType.tag == TypeTags.INTERSECTION) {
+                    elementType = ((BIntersectionType) elementType).getEffectiveType();
+                }
+
+                if (elementType.tag == TypeTags.TABLE || !isSupportedConfigType(elementType, errors, varName)) {
+                    errors.add("array element type '" + elementType + "' is not supported");
+                }
+                return true;
+            case RECORD:
+                BRecordType recordType = (BRecordType) type;
+                for (Field field : recordType.getFields().values()) {
+                    Type fieldType = field.getType();
+                    String fieldName = varName + "." + field.getName();
+                    if (!isSupportedConfigType((BType) fieldType, errors, fieldName)) {
+                        errors.add("record field type '" + fieldType + "' of field '" + fieldName + "' is not" +
+                                " supported");
+                    }
+                }
+                return true;
+            case MAP:
+                BMapType mapType = (BMapType) type;
+                if (!isSupportedConfigType(mapType.constraint, errors, varName)) {
+                    errors.add("map constraint type '" + mapType.constraint + "' is not supported");
+                }
+                return true;
+            case TABLE:
+                BTableType tableType = (BTableType) type;
+                if (!isSupportedConfigType(tableType.constraint, errors, varName)) {
+                    errors.add("table constraint type '" + tableType.constraint + "' is not supported");
+                }
+                return true;
+            case INTERSECTION:
+                return isSupportedConfigType(((BIntersectionType) type).effectiveType, errors, varName);
+            case UNION:
+                BUnionType unionType = (BUnionType) type;
+                for (BType memberType : unionType.getMemberTypes()) {
+                    if (!isSupportedConfigType(memberType, errors, varName)) {
+                        errors.add("union member type '" + memberType + "' is not supported");
+                    }
+                }
+                return true;
```

Comment:
should we return false if one of the union member types not supported? Same with table and map constraint types.